src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -247,7 +246,7 @@ public void Add(T item)
             var array = _items;
             var size = _size;
             _version++;
-            if ((uint)size < (uint)array.Length)
+            if (array.IndexInRange(size))
```

Comment:
It looks a bit ugly and it should not be on JitHelpers helpers. We should design it as if it was meant to be public API.

I am not sure whether it is a good API to add. It is way more readable to just write `size < array.Length` than to call an API to do it in a tricky way. Can we fix the JIT to morph this code into unsigned comparison instead of doing it manually? No need to introduce a new API, and no need to tech everybody to modify their code to use the new API to get the benefit.